__tests__/api.test.js

```diff
@@ -123,14 +123,54 @@ describe("GET /api/articles", () => {
         expect(articles.length).toEqual(13);
       });
   });
-  test("should be ordered by date descending", () => {
+  test("should be ordered by date descending default", () => {
     return request(app)
       .get("/api/articles")
       .then(({ body }) => {
         const { articles } = body;
         expect(articles).toBeSortedBy("created_at", { descending: true });
       });
   });
+  test("should be filtered by topic mitch", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("topic");
+        expect(articles.length).toEqual(12);
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("topic", "mitch");
+        });
+      });
+  });
+  test("should return 404 status when passed no existant topic", () => {
+    return request(app).get("/api/articles?topic=dogfood").expect(404);
+  });
+  test("should return 400 status when passed an invalid topic", () => {
+    return request(app).get("/api/articles?topic=1223").expect(404);
+  });
+  test("should be sorted by title", () => {
+    return request(app)
+      .get("/api/articles?sort_by=title")
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("title", { descending: true });
+      });
+  });
+  test("should return 404 status when passed a non existant sort_by", () => {
+    return request(app).get("/api/articles?sort_by=jeff").expect(404);
```

Comment:
I can see why you've gone for a 404 here, but we'd actually expect a 400 in this instance. We can expect a 404 with the topic, as that topic may one day exist, where as with the sort_by, it's either a correct or incorrect query that we pass into the filter 👍